### ISSUE TITLE
Brief should deal with old, long-deleted items in storage

### ISSUE THREAD

**Issue by peteroupc** [brief-rss/brief#402]
Brief keeps a record of all items that were downloaded in its storage (including the text of those items), even long after they were deleted.  Because of this, Brief's database tends to grow over time as more feed items are received, leading to a negative impact on Firefox Sync and the user experience, among other things.

Here are certain things Brief can do to address this issue:

- Periodically, purge all items older than the age given in "Remove downloaded items after...", and don't show or store items older than that age. (This could also apply to items older than the oldest article detected in the feed in question the last time it was downloaded.)
- Store a more compact version of deleted (not just trashed) items within the age limitation (for example, don't store the text of such items).
- Exempt feed items (as opposed to the list of feeds) from Firefox Sync.

**Comment by tanriol** [comment]
Yes, item deletion is not fully implemented at the moment. Sadly, I don't have enough time at the moment to fix everything properly.

This does not affect Firefox Sync performance as the items are not synced, but yes, this is currently a problem.

**Comment by peteroupc** [comment]
To avoid unnecessary database and memory usage, Brief should not _store_ any feed item if that item is older than&mdash;
- the setting given in the feed's "Remove downloaded items after..." setting, or
- the global setting "Remove downloaded items after...",

whichever is shorter.  Currently, it seems to me, when Brief refreshes a feed, it stores all its items in the database, then deletes some of the items it just stored if those items are old.  This is a wasteful use of the database and makes refreshing feed content take longer than necessary.
